presto-kinesis/src/main/java/io/prestosql/plugin/kinesis/s3config/S3TableConfigClient.java

```diff
@@ -0,0 +1,243 @@
+                    scheduler.scheduleAtFixedRate(() -> updateTablesFromS3(), 5, 600, TimeUnit.SECONDS);
+        }
+    }
+
+    /**
+     * Call S3 to get the most recent object list.
+     * <p>
+     * This is an object list request to AWS in the given "directory".
+     *
+     * @return
+     */
+    protected List<S3ObjectSummary> getObjectSummaries()
+    {
+        AmazonS3Client s3client = this.clientManager.getS3Client();
+        AmazonS3URI directoryURI = new AmazonS3URI(this.bucketUrl.get());
+
+        List<S3ObjectSummary> returnList = new ArrayList<>();
+        try {
+            log.info("Getting the listing of objects in the S3 table config directory: bucket %s prefix %s :", directoryURI.getBucket(), directoryURI.getKey());
+            ListObjectsRequest req = new ListObjectsRequest().withBucketName(directoryURI.getBucket())
+                    .withPrefix(directoryURI.getKey() + "/").withDelimiter("/").withMaxKeys(25);
+            ObjectListing result;
+
+            do {
+                result = s3client.listObjects(req);
+
+                returnList.addAll(result.getObjectSummaries());
+                req.setMarker(result.getNextMarker());
+            }
+            while (result.isTruncated());
+
+            log.info("Completed getting S3 object listing.");
+        }
+        catch (AmazonServiceException amazonServiceException) {
+            log.error(amazonServiceException.toString());
+        }
+        catch (AmazonClientException amazonClientException) {
+            log.error(amazonClientException.toString());
+        }
+
+        return returnList;
+    }
+
+    /**
+     * Connect to S3 directory to look for new or updated table definitions and then
+     * update the map.
+     */
+    protected void updateTablesFromS3()
+    {
+        long now = System.currentTimeMillis();
+
+        List<S3ObjectSummary> objectList = this.getObjectSummaries();
+        AmazonS3Client s3client = this.clientManager.getS3Client();
+        //AmazonS3URI directoryURI = new AmazonS3URI(this.bucketUrl.get());
+
+        for (S3ObjectSummary objInfo : objectList) {
+            if (!this.internalMap.containsKey(objInfo.getKey()) || objInfo.getLastModified().getTime() >= this.lastCheck) {
+                // New or updated file, so we must read from AWS
+                try {
+                    if (objInfo.getKey().endsWith("/")) {
+                        continue;
+                    }
+
+                    log.info("Getting : %s - %s", objInfo.getBucketName(), objInfo.getKey());
+                    S3Object object = s3client.getObject(
+                            new GetObjectRequest(objInfo.getBucketName(), objInfo.getKey()));
+
+                    StringBuilder resultStr = new StringBuilder("");
+                    try (BufferedReader reader = new BufferedReader(new InputStreamReader(object.getObjectContent()))) {
+                        boolean hasMore = true;
+                        while (hasMore) {
```

Comment:
You can use Guava's `CharStreams.toString` for this.

Reply:
done